prime-router/src/main/kotlin/common/HttpClientUtils.kt

```diff
@@ -0,0 +1,582 @@
+            acceptedContent: ContentType = ContentType.Application.Json,
+            timeout: Long = REQUEST_TIMEOUT_MILLIS,
+            queryParameters: Map<String, String>? = null,
+            httpClient: HttpClient? = null,
+        ): Pair<HttpResponse, String> {
+            val response = delete(
+                url = url,
+                tokens = tokens,
+                headers = headers,
+                acceptedContent = acceptedContent,
+                timeout = timeout,
+                queryParameters = queryParameters,
+                httpClient = httpClient
+            )
+            val respStr = runBlocking {
+                response.body<String>()
+            }
+            return Pair(response, respStr)
+        }
+
+        /**
+         * DELETE a resource by endpoint URL [url]
+         * A thin wrapper on top of the underlying 3rd party http client, e.g. ktor http client
+         * with:
+         * @param url: required, the url to the resource endpoint
+         * @param tokens: null default, the access token needed to call the endpoint
+         * @param headers: null default, the headers of the request
+         * @param acceptedContent: default application/json the accepted content type
+         * @param timeout: default to a system base value in millis
+         * @param queryParameters: null default, query parameters of the request
+         * @param httpClient: null default, a http client injected by caller
+         *
+         * @return the pair of response of HttpResponse and the body in string
+         *
+         */
+        fun delete(
+            url: String,
+            tokens: BearerTokens? = null,
+            headers: Map<String, String>? = null,
+            acceptedContent: ContentType = ContentType.Application.Json,
+            timeout: Long = REQUEST_TIMEOUT_MILLIS,
+            queryParameters: Map<String, String>? = null,
+            httpClient: HttpClient? = null,
+        ): HttpResponse {
+            return invoke(
+                method = HttpMethod.Delete,
+                url = url,
+                tokens = tokens,
+                headers = headers,
+                acceptedContent = acceptedContent,
+                timeout = timeout,
+                queryParameters = queryParameters,
+                httpClient = httpClient
+            )
+        }
+
+        /**
+         * Common helper for external func
+         */
+        private fun invoke(
+            method: HttpMethod,
+            url: String,
+            tokens: BearerTokens? = null,
+            headers: Map<String, String>? = null,
+            acceptedContent: ContentType? = ContentType.Application.Json,
+            timeout: Long? = REQUEST_TIMEOUT_MILLIS,
+            queryParameters: Map<String, String>? = null,
+            jsonPayload: String? = null,
+            httpClient: HttpClient? = null,
+        ): HttpResponse {
+            return runBlocking {
```

Comment:
We have a single threaded model at the moment but in general it would be better to make these functions async.

```kotlin
     private suspend fun invoke() {
     // runBlocking no longer needed and will be handled at the caller (or not if we're doing things the right way)
```

Reply:
Changing this litters the code with `runBlocking {}` calls. I suggest we cross this bridge when we get there, it seems be simpler to transition to multithread as currently written.

Reply:
I agree. It would be nice to at some point to correctly handle coroutines but this is not the ticket to do it in. I was just bringing this design decision up.